homeassistant/components/hive/climate.py

```diff
@@ -155,31 +157,30 @@ def preset_modes(self):
         return SUPPORT_PRESET
 
     @refresh_system
```

Comment:
We need to update `refresh_system` to handle a coroutine function as func parameter and await it inside the decorator.

Reply:
Not sure how that could have worked. Did you test this?

Reply:
I missed specifically testing the refresh system but I wasn't getting any errors in the logs